hlx_statics/blocks/next-prev/next-prev.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * decorates the next-prev
+ * @param {Element} block The next-prev block element
+ * 
+ */
+export default async function decorate(block) {
+
+  function extractMenuData(element) {
+    const data = [];
+
+    element.querySelectorAll('li').forEach(li => {
+      const link = li.querySelector('a');
+      const href = link ? link.getAttribute('href') : '';
+      const title = link ? link.getAttribute('title') : '';
+
+      data.push({
+        title: title,
+        href: href
+      });
+    });
+
+    return data;
+  }
+
+  const sideNavContainer = document.querySelector('.side-nav-container');
+  const menuData = extractMenuData(sideNavContainer.querySelector('ul'));
+
+  const currentPath = window?.location?.pathname;
+  function getPathFromHref(href) {
+    return href.split('http://')[1] || '';
+  }
+
+  const currentPathData = menuData.find(menu => `/${getPathFromHref(menu.href)}` === currentPath);
+  const currentIndex = menuData.findIndex(menu => menu.href === currentPathData?.href);
+
+  const previousPage = currentIndex > 0 ? menuData[currentIndex - 1] : null;
+  const nextPage = currentIndex < menuData.length - 1 ? menuData[currentIndex + 1] : null;
+
+  const Prev = block.querySelector('.next-prev>div>div');
+  Prev.classList.add("spectrum-Body", "spectrum-Body--sizeM");
+  block.appendChild(Prev);
+
+  const Prev_one = document.createElement('div');
+  Prev_one.classList.add('main-one-innerDiv');
+  Prev.appendChild(Prev_one);
+
+  const createIconSvg = (direction) => {
+    const path = direction === 'left'
+      ? 'M12 18a1.988 1.988 0 0 0 .585 1.409l7.983 7.98a2 2 0 1 0 2.871-2.772l-.049-.049L16.819 18l6.572-6.57a2 2 0 0 0-2.773-2.87l-.049.049-7.983 7.98A1.988 1.988 0 0 0 12 18z'
+      : 'M24 18a1.988 1.988 0 0 1-.585 1.409l-7.983 7.98a2 2 0 1 1-2.871-2.772l.049-.049L19.181 18l-6.572-6.57a2 2 0 0 1 2.773-2.87l.049.049 7.983 7.98A1.988 1.988 0 0 1 24 18z';
+    return `<svg viewBox="0 0 36 36" focusable="false" aria-hidden="true" role="img" class="spectrum-Icon spectrum-Icon--sizeM"><path d="${path}"></path></svg>`;
+  }
+
+  const createLink = (href, title, direction, classname) => {

```

Comment:
There's a [Link component](https://github.com/adobe/adobe-io-website/blob/fresh-spectrum-less/hlx_statics/components/link.js) (based on [aio-theme's Link](https://github.com/adobe/aio-theme/blob/main/packages/gatsby-theme-aio/src/components/Link/index.js#L18)) that's very similar to this that's being used by [Heading](https://github.com/adobe/adobe-io-website/blob/fresh-spectrum-less/hlx_statics/components/heading.js#L39). Could we refactor this to reuse that?

In general, if there's a block that's being used by other blocks, (like Link where it's used by [Heading](https://github.com/adobe/aio-theme/blob/main/packages/gatsby-theme-aio/src/components/Heading/index.js#L114) and [NextPrev](https://github.com/adobe/aio-theme/blob/main/packages/gatsby-theme-aio/src/components/NextPrev/index.js#L33)) I think we should create it under components to facilitate code reuse. Reason for component, rather than block is because EDS doesn't allow blocks to be nested within blocks. Please let me know what you think @timkim

Reply:
Ya, let's reuse the link component here. Should be pretty straight forward to import the link into here.

Reply:
I've resolved it